### PR TITLE
Update and focus "REST API" in Architecture doc

### PR DESCRIPTION
**Pull request by chenette**
- Remove API details that aren't appropriate for an architecture guide (plus, they're out of date)
- Add link to the REST API Reference (which has the current API information)

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: docs/source/architecture/rest_api.rst [hunk 1]
```diff
@@ -2,313 +2,47 @@
...
-Endpoints
-=========
-
-The endpoints include RESTful references to resources stored in the Sawtooth
+The API endpoints include RESTful references to resources stored in the Sawtooth
 ledger that clients might be interested in, like blocks and transactions, as
 well as RESTish metadata, like batch status.
 
+All endpoints that return lists of resources will automatically paginate the
+results, up to a default limit of 1000 resources.
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply docs/source/architecture/rest_api.rst]: https://github.com/hyperledger/sawtooth-core/blob/master/validator/sawtooth_validator/state/client_handlers.py#L55

Yep, it's 100

File: docs/source/architecture/rest_api.rst [hunk 2]
```diff
@@ -2,313 +2,47 @@
...
-that completely documents every implemented aspect of the API, is both human and
-machine readable, and can be compiled by a variety of toolsets (including for
-this Sphinx-based document). The spec file itself can be found
-`here <https://github.com/hyperledger/sawtooth-core/blob/master/rest_api/openapi.yaml>`_.
-
-
-HTTP Status Codes
+Hyperledger Sawtooth provides a REST API (see the :doc:`../rest_api`) that
+allows clients to interact with a validator using common HTTP/JSON standards.
+It is a pragmatic RESTish API that provides a language-neutral interface for
```
Comment (aludvik) [inline docs/source/architecture/rest_api.rst]: What is the motivation for calling this RESTish? Is it accurate enough to call it RESTful?

File: docs/source/architecture/rest_api.rst [hunk 3]
```diff
@@ -2,313 +2,47 @@
...
+submitting transactions and reading blocks.
+
+The Sawtooth REST API is intended only as a simple interface for client use. The
+validator and transaction processors use a ZMQ/Protobuf interface [LINK]
+that is more efficient and robust, as well as slightly more complicated.
+Note that clients also have the option of using the ZMQ/Protobuf interface to
+communicate with the validator.
+
+Note that the REST API is a lightweight layer on top of Sawtooth's internal ZMQ
```
Comment (aludvik) [inline docs/source/architecture/rest_api.rst]: You start two sentences in a row with "Note that".

File: docs/source/architecture/rest_api.rst [hunk 4]
```diff
@@ -2,313 +2,47 @@
...
-
-
-HTTP Status Codes
+Hyperledger Sawtooth provides a REST API (see the :doc:`../rest_api`) that
+allows clients to interact with a validator using common HTTP/JSON standards.
+It is a pragmatic RESTish API that provides a language-neutral interface for
+submitting transactions and reading blocks.
+
+The Sawtooth REST API is intended only as a simple interface for client use. The
+validator and transaction processors use a ZMQ/Protobuf interface [LINK]
```
Comment (aludvik) [inline docs/source/architecture/rest_api.rst]: I think you need to either remove "[LINK]" or fill it in with an actual link.

File: docs/source/architecture/rest_api.rst (hunk 2 above)
Comment (chenette) [reply docs/source/architecture/rest_api.rst]: The original doc said "RESTish".  However, I can't discern a difference between these two terms. Also, Peter just said it's OK to change it to "RESTful". So I shall make this change.

File: docs/source/architecture/rest_api.rst (hunk 3 above)
Comment (chenette) [reply docs/source/architecture/rest_api.rst]: :-(  I'll fix that.

File: docs/source/architecture/rest_api.rst [hunk 5]
```diff
@@ -2,313 +2,47 @@
...
+in an envelope that may include the requested resource, chain head ID, a link
+to the requested resource, and paging information (described below).
 
+The JSON response envelope also includes the details needed to parse errors. For
+more information, see the :doc:`../rest_api/error_codes`.
 
 Pagination
-----------
-All endpoints that return lists of resources will automatically paginate
-results, limited by default to 1000 resources. In order to specify what range of
```
Comment (peterschwarz) [inline docs/source/architecture/rest_api.rst]: This should still 100

File: docs/source/architecture/rest_api.rst [hunk 6]
```diff
@@ -2,313 +2,47 @@
...
-.. list-table::
-   :widths: 8, 72
-
-   * - **head**
-     - The id of the block to use as the chain head. This is particularly
-       useful to request older versions of state *(defaults to the latest chain
-       head)*.
-   * - **count**
-     - For paging, this item specifies the number of resources to fetch
-       *(defaults to 1000)*.
```
Comment (peterschwarz) [inline docs/source/architecture/rest_api.rst]: This should be 100

File: docs/source/architecture/rest_api.rst (hunk 5 above)
Comment (chenette) [reply docs/source/architecture/rest_api.rst]: I made the change to 100. Your comment is on the deleted content. Please see the "View" version of rest_api.rst. https://github.com/chenette/sawtooth-core/blob/0f0bb5a81d540810ce38031c7cd1ddf02f332a7b/docs/source/architecture/rest_api.rst

File: docs/source/architecture/rest_api.rst (hunk 6 above)
Comment (chenette) [reply docs/source/architecture/rest_api.rst]: This comment is on deleted content that doesn't exist in the current version.

File: docs/source/architecture/rest_api.rst (hunk 5 above)
Comment (peterschwarz) [reply docs/source/architecture/rest_api.rst]: I'm bad, this is deleted code.

File: docs/source/architecture/rest_api.rst (hunk 6 above)
Comment (peterschwarz) [reply docs/source/architecture/rest_api.rst]: I'm bad, this is deleted code.

**Pull request merged**